### PR TITLE
feat(dialog-routing): Open dialog through routes

### PR DESCRIPTION
**Pull request by guillerot**
## Description

Provide component, util function, and models to open dialog through routes

-----

https://github.com/LuccaSA/lucca-front/assets/2039493/cf24ea16-af6b-48b4-add3-53aa255e7a4d

-----

### CODE REVIEW

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2901/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/18951/ui-diff)

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2901/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/18952/ui-diff)

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/18955/ui-diff)

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2901/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/18957/ui-diff)

**Review by Supamiu — APPROVED** [review]

**Review by GuillaumeNury — DISMISSED** [review]

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2901/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/19789/ui-diff)

File: packages/ng/dialog/dialog-routing/dialog-routing.utils.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import { InjectionToken } from '@angular/core';
+import { Route } from '@angular/router';
+import { DialogRoutingComponent } from './dialog-routing.component';
+import { DialogRouteConfig } from './dialog-routing.models';
+import { LuDialogConfig, LuDialogData } from '../model';
+import { ComponentType } from '@angular/cdk/portal';
+import { firstValueFrom, isObservable, Observable } from 'rxjs';
+
+export type Deferrable<T> = Promise<T> | Observable<T> | T;
```
Comment (g-drouard) [inline packages/ng/dialog/dialog-routing/dialog-routing.utils.ts:9]: On peut utiliser `MayBeAsync` fournit par angular (c'est exactement le même typage)

**Review by Supamiu — APPROVED** [review]

**Review by jeremie-lucca — APPROVED** [review]

**Pull request merged**